### PR TITLE
[Profiler] add missing field device_resource_id

### LINKED ISSUE #121479: `device_resource_id` is missing in `FunctionEvent`
### 🐛 Describe the bug

I would like to get the stream ids from profiled events, however, I found that

`device_resource_id` defined here

https://github.com/pytorch/pytorch/blob/e628f2cc669b711ce96795af65b55e06fa7d38f3/torch/csrc/autograd/init.cpp#L236-L239

is missing in the definition of `FunctionEvent`.

### Versions

The latest version.

cc @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7 @robieta @chaekit @aaronenyeshi @nbcsm @guotuofeng @guyang3532 @gaoteng-git @tiffzhaofb @dzhulgakov @davidberard98

### PR DESCRIPTION
**Pull request by lucifer1004**
Fixes #121479 

cc @robieta @chaekit @aaronenyeshi @nbcsm @guotuofeng @guyang3532 @gaoteng-git @tiffzhaofb @dzhulgakov @davidberard98

### CODE REVIEW

**Comment by davidberard98** [comment]
@lucifer1004 thanks for the PR! can you take a look at the linter failures? (also, I rebased the PR past some failures, so you might have to pull again when you check out your code. You can develop off of the `viable/strict` branch to avoid some of these preexisting failures)

**Review by aaronenyeshi — APPROVED** [review]
LGTM!

**Review by aaronenyeshi — CHANGES_REQUESTED** [review]
Please remove the update to third_party/XNNPACK, looks like this may be accidentally picked up in your change. Also, please check lintrunner error for torch/autograd/profiler_util.py

File: torch/autograd/profiler_util.py [hunk 1]
```diff
@@ -485,6 +486,9 @@ def __init__(
         self.sequence_nr: int = sequence_nr
         self.device_type: DeviceType = device_type
         self.device_index: int = device_index
+        self.device_resource_id: int = (
+thread if device_resource_id is None else device_resource_id
```
Comment (aaronenyeshi) [inline torch/autograd/profiler_util.py:490]: This need proper indentation, around 12 spaces.
```suggestion
            thread if device_resource_id is None else device_resource_id
```

**Comment by lucifer1004** [comment]
I do see the difference via `git diff`, but cannot edit it since it is a directory.

**Review by aaronenyeshi — APPROVED** [review]
LGTM!

**Pull request closed without merging**